src/TypeDefinition.js

```diff
@@ -290,23 +290,18 @@ export type NavigationAction =
   | NavigationStackAction
   | NavigationTabAction;
 
-export type NavigationScreenRouteConfig = {
-  /** React component or navigator to render for this route */
-  screen: NavigationScreenComponent<*> | NavigationNavigator<*>,
-  navigationOptions?: NavigationScreenOptions,
-  path?: string,
-};
-
-export type NavigationLazyScreenRouteConfig = {
-  /** React component or navigator to lazily require and render for this route */
-  getScreen: () => (NavigationScreenComponent<*> | NavigationNavigator<*>),
+export type NavigationRouteConfig<T> = T & {
   navigationOptions?: NavigationScreenOptions,
   path?: string,
 };
 
-export type NavigationRouteConfig =
-  | NavigationScreenRouteConfig
-  | NavigationLazyScreenRouteConfig;
+export type NavigationScreenRouteConfig = NavigationScreenRouteConfig<{
+  // React component or navigator for this route */
+  screen: NavigationComponent,
+} | {
+  // React component to lazily require and render for this route */
```

Comment:
So, the component deferred with `getScreen` is not supposed to be a navigator/NavigationComponent (have a router).

This type makes that unclear. If this is needed to make flow pass, then I think its fine for now, but otherwise we should probably change this to be a normal react component that receives a `NavigationScreenProp`